pressbooks/pressbooks
# Pb 2261 cover image tweaks

This PR fixes #2261 by changing the instructions a user sees when uploading a cover image via the Book Info page. It also checks the file dimensions and aspect ratio for the image a user uploads. If the conditions are not satisfied, an error message is displayed and the new cover image is not uploaded/saved. **Note:** any other changes to book info should still be saved/updated successfully, even if the cover image is rejected for size reasons. This PR also downsizes any images with valid aspect ratios larger than 1500x1500px before uploading/saving.

To test:
1. Attempt to upload a cover image with a height of <800px. Note that the image is not saved and an error message is displayed.
![too_small](https://user-images.githubusercontent.com/13485451/126012166-65426d30-0c11-497b-86f7-9e1cf429674e.jpg)
2. Upload an image with an aspect ratio of <.66. Note that the image is not saved and an error message is displayed.
![too_wide](https://user-images.githubusercontent.com/13485451/126012432-d0410799-d04c-4d02-9c2d-542a44b65a68.jpg)
3. Upload an image with an aspect ratio of >1. Note that the image is not saved and an error message is displayed.
![too_tall](https://user-images.githubusercontent.com/13485451/126012077-abe7bd9d-1f04-4954-8486-464a511ddd75.jpg)
4. Upload a very large image with an acceptable aspect ratio. Not that the image is saved and has been resized to a maximum height of 1500px (confirm this by visiting the book's media and inspecting the source image dimensions).
![large_test](https://user-images.githubusercontent.com/13485451/126011985-bbc97db4-8962-4bf2-8ed6-245192de4853.jpg)
